src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx

```diff
@@ -45,13 +60,13 @@ const CalculationConfig = (props) => {
     if (diff.embeddingSettings) {
       // If this is an embedding change, indicate to user that their changes are not
       // applied until they hit Apply.
-      setChangesOutstanding(true, () => {
-        dispatch(updateProcessingSettings(
-          experimentId,
-          FILTER_UUID,
-          diff,
-        ));
-      });
+
+      setChangesOutstanding(true);
+      dispatch(updateProcessingSettings(
+        experimentId,
+        FILTER_UUID,
+        diff,
+      ));
```

Comment:
Dispatch is moved out so that it'll be called. `setState` does not have a callback,

Reply:
You're right, this is a bug. Keep in mind, though: there is no guarantee that `changesOutstanding` will be set *before* the dispatch is called, as the setter for `useState` is asyc.

If the timing of the two events (`setChangesOutstanding` & `dispatch`) does not matter, you can keep it as is. Otherwise, they should be linked via `useEffect`.

Reply:
Yes, I'm aware of the racing of the two conditions. Ideally 

As they involve components only in the browser (state and Redux), if everything goes well, they should resolve in a couple of milliseconds. That should be enough time for the states to resolve before the user can click Apply. However, as you said, there is no guarantee.

Ideally, `setChangesOutstanding` is fired after dispatch finishes changing. So, I think it'll be better to implement it with `useEffect`?

Reply:
I can fix this by adding a variable under `useRef` to skip the first run and `setChangesOutstanding` using `useEffect`. However, that causes 2 tests to not be able to run, because then the Alert requires Redux store change (which is not yet supported by `mockStore`). What do you think?

Reply:
I agree, for the purposes of this it doesn't matter. We can ignore it for now.